src/components/dialogs/parameters/dynamicsimulation/curve/dialog/model-filter.tsx

```diff
@@ -0,0 +1,310 @@
+
+interface ModelVariableDefinitionInfos {
+    name: string;
+    unit: string;
+}
+
+interface DynamicSimulationModelBack {
+    modelName: string;
+    equipmentType: EQUIPMENT_TYPES;
+    variableDefinitions: ModelVariableDefinitionInfos[];
+    variablesSets: { name: string; variableDefinitions: ModelVariableDefinitionInfos[] }[];
+}
+
+interface DynamicSimulationModel {
+    name: string;
+    equipmentType: EQUIPMENT_TYPES;
+}
+
+export interface GetSelectedVariablesHandle {
+    api: {
+        getSelectedVariables: () => ModelVariable[];
+    };
+}
+
+interface ModelFilterProps {
+    equipmentType: EQUIPMENT_TYPES;
+}
+export type ModelVariable = {
+    id: string;
+    name: string;
+    parentId?: string;
+    variableId?: string;
+};
+
+const modelsToVariablesTree = (models: DynamicSimulationModelBack[]) => {
+    return models.reduce<Record<string, Record<string, string | Record<string, string>>>>(
+        (obj, model) => ({
+            ...obj,
+            [model.modelName]: {
+                ...model.variableDefinitions.reduce(
+                    (obj, variable) => ({
+                        ...obj,
+                        [variable.name]: variable.name,
+                    }),
+                    {}
+                ),
+                ...model.variablesSets.reduce(
+                    (obj, variablesSet) => ({
+                        ...obj,
+                        [variablesSet.name]: variablesSet.variableDefinitions.reduce(
+                            (obj, variable) => ({
+                                ...obj,
+                                [variable.name]: variable.name,
+                            }),
+                            {}
+                        ),
+                    }),
+                    {}
+                ),
+            },
+        }),
+        {}
+    );
+};
+
+const variablesTreeToVariablesArray = (variablesTree: Record<string, any>, parentId?: string) => {
+    let result: ModelVariable[] = [];
+    Object.entries(variablesTree).map(([key, value]) => {
+        const id = parentId ? `${parentId}/${key}` : key;
+        if (typeof value === 'object') {
+            // make container element
+            result = [
+                ...result,
+                {
+                    id: id,
+                    name: key,
+                    parentId: parentId,
+                },
+            ];
+            // make contained elements
+            result = [...result, ...variablesTreeToVariablesArray(value, id)];
+        }
+        if (typeof value === 'string') {
+            result = [
+                ...result,
+                {
+                    id: id,
+                    name: value,
+                    parentId: parentId,
+                    variableId: key,
+                },
+            ];
+        }
+        return result;
+    });
+
+    return result;
+};
+
+const makeGetModelLabel = (intl: IntlShape) => (value: string) =>
```

Comment:
this function is called only one time with simple code. maybe removed and call intl.formatMessage directly

Reply:
Theoretically, YES, but in my opinion, it is better to group with makeGetVariableLabel at the same place. Two methods manage prefixes while generating dynamic key.